### PR TITLE
Enable automated Github release

### PR DESCRIPTION
**Pull request by pakrym-stripe**
Turns the github workflow from manual to automatic on tag push.

### CODE REVIEW

File: src/Stripe.net/Stripe.net.csproj [hunk 1]
```diff
@@ -4,8 +4,8 @@
   <PropertyGroup>
     <Description>Stripe.net is a sync/async .NET 4.6.1+ client, and a portable class library for the Stripe API.  (Official Library)</Description>
     <AssemblyTitle>Stripe.net</AssemblyTitle>
-    <VersionPrefix>39.107.0</VersionPrefix>
     <Version>39.107.0</Version>
+    <VersionPrefix>$(Version)</VersionPrefix>
```
Comment (pakrym-stripe) [inline src/Stripe.net/Stripe.net.csproj:8]: Fewer places to update.

Comment (dcr-stripe) [inline src/Stripe.net/Stripe.net.csproj:8]: Nice.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,12 +1,11 @@
 name: CI
 
 on:
-  workflow_dispatch: {}
```
Comment (dcr-stripe) [inline .github/workflows/ci.yml:4]: Do we want to keep this in case we need to manually publish?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
   push:
     branches:
       - master
     tags:
-      - '*'
+      - v*
```
Comment (dcr-stripe) [inline .github/workflows/ci.yml:8]: This can't be a more strict regex right?

**Review by dcr-stripe — COMMENTED** [review]
🎉 awesome! Just some small comments and then good to 🚢 !

File: .github/workflows/ci.yml (hunk 3 above)
Comment (pakrym-stripe) [reply .github/workflows/ci.yml:8]: We can but I'm not sure if it will add anything.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pakrym-stripe) [reply .github/workflows/ci.yml:4]: Good idea.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dcr-stripe) [reply .github/workflows/ci.yml:9]: I think it'd be good just to make sure this doesn't accidentally get released for some other tag. Better to be overly cautious here.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -69,7 +69,8 @@ jobs:
           path: nuget/
 
   publish:
-    if: ((github.event_name == 'workflow_dispatch') && (github.ref == 'refs/heads/master') && endsWith(github.actor, '-stripe'))
+    name: Publish
+    if: (((github.event_name == 'push') || (github.event_name == 'workflow_dispatch')) && startsWith(github.ref, 'refs/tags/v') && endsWith(github.actor, '-stripe'))
```
Comment (dcr-stripe) [inline .github/workflows/ci.yml:73]: Double checking here that we're OK with having workflow_dispatch only work on ref/tags/ rather than master

**Review by dcr-stripe — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yml (hunk 4 above)
Comment (pakrym-stripe) [reply .github/workflows/ci.yml:73]: Yep, now that the workflow is going to be present in tag commit we should be good.

**Pull request merged**